bst_limits.c

```diff
@@ -39,7 +41,7 @@ static int parse_complete_rlim_t(int resource, rlim_t *value, char const *value_
 
 	/* Before casting to rlim_t, check against the max possible rlim_t
 	   value so that the (rlim_t) cast is guaranteed correct. */
-	if ((uintmax_t)parse_val >= RLIM_INFINITY) {
+	if ((uintmax_t)parse_val >= BST_RLIM_INFINITY) {
```

Comment:
We do essentially need to respect the values that are accepted by `setrlimit` and the end of the day, though, otherwise we'll have to figure it out when we're passing nonsense values into `setrlimit`?

Reply:
If we're passing nonsense to setrlimit, we'll just get EINVAL and abort, which seems right. Syscall interfaces (apart from actual syscall numbers and intptr sizes between 32 and 64-bit) are fairly stable across platforms, so our definition of RLIM_INFINITY should hold fairly well. It's basically -1.